MSAL/src/native_auth/controllers/MSALNativeAuthBaseController.swift

```diff
@@ -149,12 +149,20 @@ class MSALNativeAuthBaseController {
         completion(response, error)
     }
 
-    func performRequest<T>(_ request: MSIDHttpRequest, context: MSIDRequestContext) async -> Result<T, Error> {
+    func performRequest<T: MSALNativeAuthResponseCorrelatable>(
+        _ request: MSIDHttpRequest,
+        context: MSALNativeAuthRequestContext
+    ) async -> Result<T, Error> {
         return await withCheckedContinuation { continuation in
             request.send { result, error in
-                if let error = error {
+                if let error = error, let errorWithCorrelationId = error as? MSALNativeAuthResponseCorrelatable {
```

Comment:
I'd add an inner if just for the errorWithCorrelationId so we don't need to repeat the code at line 160 also at line 163. 
```suggestion
                if let error = error {
                    if let errorWithCorrelationId = error as? MSALNativeAuthResponseCorrelatable {
                        context.setServerCorrelationId(errorWithCorrelationId.correlationId)
                    }
                    continuation.resume(returning: .failure(error))
                } else if let response = result as? T {
```

Reply:
ok, done